### PR TITLE
Status filter 390

### PR DESCRIPTION
**Pull request by hakhalid11**
[Link to ZenHub issue.](https://app.zenhub.com/workspaces/doj-crt-intake-scrum-board-5d03bf56c1c8a35d482eca0f/issues/18f/crt-portal/390undefined)

## What does this change?
-adds checkbox for status selection.
-adds on load default status selection
## Screenshots (for front-end PR):

image

image

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.)
+ [x] Check for [accessibility](/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.)
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/forms.py [hunk 1]
```diff
@@ -901,6 +900,9 @@ class Filters(ModelForm):
         })
     )
 
+    def __init__(self, *args, **kwargs):
+        ModelForm.__init__(self, *args, **kwargs)
```
Comment (cantsin) [inline crt_portal/cts_forms/forms.py:904]: :thinking: Any particular reason we need this? I removed and seems to work fine ...

File: crt_portal/cts_forms/forms.py [hunk 2]
```diff
@@ -1009,7 +1011,9 @@ def __init__(self, *args, **kwargs):
         self.fields['assigned_section'] = ChoiceField(
             widget=ComplaintSelect(
                 label='Section',
-                attrs={'class': 'usa-select text-bold text-uppercase crt-dropdown__data'},
+                attrs={
+                    'class': 'usa-select text-bold text-uppercase crt-dropdown__data'
+                },
```
Comment (cantsin) [inline crt_portal/cts_forms/forms.py:1016]: Any reason `attrs` was automatically formatted? In general if this is not part of a cleanup then I'd leave this alone, as it adds to PR noise.

File: crt_portal/cts_forms/templates/landing.html [hunk 3]
```diff
@@ -132,7 +132,7 @@ <h2 id="about-the-division" class="h1__display">{% trans "About the Civil Rights
               <p class="crt-landing--columns crt-landing--largetext">{% trans "We sue or prosecute individuals and organizations who violate civil rights laws." %}</p>
               <p>{% trans "You can help us do this work by reporting a possible civil rights violation through our" %} <a href="{% url 'crt_report_form' %}">{% trans "online form" %}</a>.</p>
             </div>
-            <div class="tablet:grid-col-4">
+            <div class="tablet:grid-col-4 crt_landing--section_item">
```
Comment (cantsin) [inline crt_portal/cts_forms/templates/landing.html:135]: All the changes in this file seem unrelated?

File: crt_portal/static/js/complaint_view_filters.js [hunk 4]
```diff
@@ -223,6 +224,25 @@
...
+  checkBoxView.getValues = function(el) {
+    if (el.checked) {
+      filterDataModel[event.target.name].push(el.value);
+      return el.value;
+    } else {
+      var index = filterDataModel[event.target.name].indexOf(el.value);
+      filterDataModel[event.target.name].splice(index, 1);
+      return el.value;
+    }
+  };
```
Comment (cantsin) [inline crt_portal/static/js/complaint_view_filters.js:244]: Did we test in IE11/Edge? If not, I can do this myself, just wanted to verify that this code would work. I don't think anything here will break, but its always good to test to make sure ... (I've been surprised before)

File: crt_portal/static/js/complaint_view_filters.js [hunk 5]
```diff
@@ -223,6 +224,25 @@
...
+
+  checkBoxView.getValues = function(el) {
+    if (el.checked) {
+      filterDataModel[event.target.name].push(el.value);
+      return el.value;
+    } else {
+      var index = filterDataModel[event.target.name].indexOf(el.value);
+      filterDataModel[event.target.name].splice(index, 1);
+      return el.value;
+    }
```
Comment (cantsin) [inline crt_portal/static/js/complaint_view_filters.js:243]: Maybe pull out `return el.value;` here?

File: crt_portal/static/js/complaint_view_filters.js (hunk 4 above)
Comment (hakhalid11) [reply crt_portal/static/js/complaint_view_filters.js:242]: I don't have IE on my mac yet, If you don't mind giving it a quick run through on IE/Edge that would be helpful

**Comment by cantsin** [comment]
Bringing out our conversation in the open, IE11 does not work:



We will need to polyfill `forEach` on `NodeList`.

**Comment by hakhalid11** [comment]
> Bringing out our conversation in the open, IE11 does not work:
> 
> 
> 
> We will need to polyfill `forEach` on `NodeList`.

Thanks James. I fixed the IE11 issue. Further tests on IE11 and edge came out clean. I think this should be good to go

File: crt_portal/cts_forms/templates/forms/complaint_view/index/filter-controls.html [hunk 6]
```diff
@@ -2,15 +2,19 @@
...
+> 
+  <div class="intake-filters">
+    {% include 'forms/complaint_view/index/filters/checkbox_status_filter.html' %}
+    <button class="usa-button " type="submit">Apply Filters</button>
+  </div>
   <div class="intake-filters">
     {% include 'forms/complaint_view/index/filters/section_filter.html' %}
     {% include 'forms/complaint_view/index/filters/contact_filter.html' %}
     {% include 'forms/complaint_view/index/filters/incident_location_filter.html' %}
-    {% include 'forms/complaint_view/index/filters/status_filter.html' %}
```
Comment (cantsin) [inline crt_portal/cts_forms/templates/forms/complaint_view/index/filter-controls.html:13]: Ooh we need to remove this file. It's not used now, right?

File: crt_portal/static/sass/custom/intake.scss [hunk 7]
```diff
@@ -95,6 +98,19 @@
...
+      flex-wrap: wrap;
+      white-space: normal;
+  
+      li {
+        margin: 0.5rem;
+      }
+      + .usa-button {
+        margin-left: 2rem;
+      }
+    }
```
Comment (cantsin) [inline crt_portal/static/sass/custom/intake.scss:112]: Did we inadvertently change something here? The CSS looks off now:

Plus the dropdown assignee doesn't work for me now.

Comment (cantsin) [reply crt_portal/static/sass/custom/intake.scss:115]: @hakhalid11 thank you for fixing the assignee dropdown! 

The CSS still looks a bit off to me. Here's what it looks like in `develop`:

**Comment by cantsin** [comment]
@hakhalid11 CI tests fail -- 

` • Error: Failed action: no element matching selector "#submitted-sort"`

I think you'll have to adjust the pa11y script here. Almooooost there.
